Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/dialogs/GamePropertiesDialog.java

```diff
@@ -112,10 +116,34 @@ private void clearGameSettings(String gameId)
         Toast.makeText(getContext(), "Unable to clear settings for " + gameId,
                 Toast.LENGTH_SHORT).show();
       }
+      mHadGameProfiles = false;
     }
     else
     {
       Toast.makeText(getContext(), "No game settings to delete", Toast.LENGTH_SHORT).show();
     }
   }
+
+  private boolean recursivelyDeleteGameProfiles(File file, String gameId)
+  {
+    if (file.isDirectory())
+    {
+      try
+      {
+        for (File child : file.listFiles())
+        {
+          if (child.getName().startsWith(gameId) && child.isFile())
+          {
+            child.delete();
+            mHadGameProfiles = true;
+          }
+          recursivelyDeleteGameProfiles(child, gameId);
+        }
+      }
+      catch (Exception ignored)
```

Comment:
I forgot to mention this earlier. This should catch the specific exception you need to catch instead of just `Exception`, to make it clearer what the purpose of this `catch` is. (I'm imagining it's something like `IOException`, but I'm not sure.)

Reply:
Android studio complains that `IOException` can't be raised by this code block. I think I just need `NullPointerException` since `file.listFiles()` could potentially cause that.

I'll get better at recognizing these exceptions eventually.

Reply:
I don't really see how `file.listFiles()` could cause a `NullPointerException`, though... You're only ever passing non-null `File`s into the function.

Why was this `catch` added to begin with? Because of an observed crash?

Reply:
It was an Android Studio complaint. I also suspected that the exception wouldn't happen but I wasn't 100% sure. Should I just suppress the warning instead?

Reply:
In that case, I would suggest using the `@NotNull` annotation if you care about getting rid of the warnings.

Reply:
`@NonNull` alone wasn't enough to take care of the warnings but I think I found a way. I also applied the same change to `DirectoryInitialization.deleteDirectoryRecursively`.